### DISCUSSION TITLE
Panopticon auto add

### DISCUSSION THREAD

**Discussion by vinespie** [akeeba/panopticon#338] [Q&A (Community Support)] [answered]
Is there a solution to automate the addition of a site in Panopticon?

I use a backup made with Akeeba Backup and the UNiTE script to install sites. As I install a lot of them, I would like to avoid going to each site to take the configuration from the connector.

**Comment by nikosdion** [comment]
Not yet, [but we're getting there](https://github.com/akeeba/panopticon/issues/153).

**Reply by vinespie** [reply]
Are there plans to add an API allowing you to edit the list of sites ?

**Reply by nikosdion** [reply]
I am not sure if it would make a lot of sense. What is your use case which would be met by an API but not CLI commands?

**Reply by vinespie** [reply]
I created a program allowing you to perform the complete installation of Joomla (web server, joomla, etc.). In this process I want to integrate the addition into Panopticon. This can only be done through an API, the servers are not the same. The advantage of the API is that any program or website can consume it. I have around 2000 sites to integrate and the API would be very useful.

**Reply by nikosdion** [reply, marked as answer]
An API would be useful in this case, but it would also be a mid- to long-term goal. There's only one of me, nobody else seems interested in contributing code, I only have 24 hours in a day, and I don't make any money out of this. Between all these restrictions, an API would be pretty low in the implementation list. If anyone wants to contribute code, by all means.

In the meantime, you could use the CLI commands with something like Ansible, or just produce raw data. The site configuration is just JSON data and its format [is documented](https://github.com/akeeba/panopticon/wiki/Site-Parameters).

**Reply by vinespie** [reply]
Yes, I understand that it is not a priority. If I move forward on my own, I will make a contribution. Thanks for the help.

**Reply by nikosdion** [reply]
My initial thoughts on the API implementation are on https://github.com/akeeba/panopticon/issues/344. You can track that to see the progress in the (hopefully not too distant!) future ;)
